### PR TITLE
GSOC-2015 Julia application

### PR DESCRIPTION
**Pull request by SSchamnad**
I have provided my proposal for GSOC-2015. Your inputs would be appreciated

### CODE REVIEW

File: 2015/proposals/schamnad-sanjeed.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Title
```
Comment (r-gaia-cs) [inline 2015/proposals/schamnad-sanjeed.md]: Use

~~~
# JuliaWeb: A secure Web framework for the Julia Programming Language
~~~

File: 2015/proposals/schamnad-sanjeed.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+juliaTLS API: The different modules will be combined to create Julia native atomic functions to provide TLS functionalitiy. A full TLS negotiation will be tested via HTTPS, with a simple set of methods (get, post). 
+
+### August 17th - August 21th 19:00 UTC
+
+**Week to scrub code, write tests, improve documentation, etc.**
+
+## Future works
+The juliaTLS package is intended to provide the basis upon which the secure Julia implementations of various application layer protocols can be built. The various cryptographic libraries may be extended to provide additional capabilities and functionality.
+
+## Open Source Development Experience
```
Comment (r-gaia-cs) [inline 2015/proposals/schamnad-sanjeed.md]: Do you have any pull request merged at one Julia project that is a wrap of a C library?

File: 2015/proposals/schamnad-sanjeed.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+The output of this framework will be an encrypted stream.
+
+Clients and Peers will be authenticated using the Certificate authentication methodology, in particular by using X.509 certificates. This will be based on the methodology incorporated by OpenSSL, in particular the following functions:
+* openssl req -new -key privkey.pem -out cert.csr
+* int SSL_CTX_use_certificate(SSL_CTX *ctx, X509 *x);
+* X509 *TLS_CTX_get0_certificate(const SSL_CTX *ctx);
+* X509 *TLS_get_certificate(const SSL *ssl);
+
+3) The Application Programming Interface (API) layer
```
Comment (r-gaia-cs) [inline 2015/proposals/schamnad-sanjeed.md]: I believe that other languages, e.g. Python or Ruby, has API for OpenSSL/TLS. Did you check it for inspiration?

**Comment by r-gaia-cs** [comment]
@SSchamnad Thanks for your proposal.

**You need to submit your proposal to https://www.google-melange.com/gsoc/homepage/google/gsoc2015 as soon as possible.** The deadline is March 27th 19:00 UTC.

Do you have a possible mentor?

File: 2015/proposals/schamnad-sanjeed.md (hunk 2 above)
Comment (SSchamnad) [inline 2015/proposals/schamnad-sanjeed.md]: Currently No. I had mentioned my lack of experience to Seth when I was discussing the project with him.
 
I have experience in working with the Web Stack but not native to Julia, but I have already started learning how to write wrappers from a few resources ( http://docs.julialang.org/en/latest/manual/calling-c-and-fortran-code/). I have also discussed the project with Amit Murthy who has developed the julia wrappers for the OpenSSL crypto library, and Seth is always helpful.

File: 2015/proposals/schamnad-sanjeed.md (hunk 1 above)
Comment (SSchamnad) [inline 2015/proposals/schamnad-sanjeed.md]: I have edited and committed my changes

File: 2015/proposals/schamnad-sanjeed.md (hunk 3 above)
Comment (SSchamnad) [inline 2015/proposals/schamnad-sanjeed.md]: I did.
I went through the Python API for OpenSSL. But my research was not limited to them, I had to go to the original sources of these implementations. I read the RFC's of TLS and the respective implementations in C/C++ by OpenSSL and gnuTLS while working on my proposal.

**Comment by SSchamnad** [comment]
Thank you for your inputs Raniere.
I have submitted my proposal on GSOC's homepage. I submitted my proposal by copy-pasting the text from the Word document instead of going the pandoc route outlined in the default instructions. I hope that won't cause any problems.

I have been working with Seth Bromberger from the time I began working on this proposal, he's been really helpful. I would like to work under him if my project proposal got selected.

**Comment by r-gaia-cs** [comment]
@SSchamnad Great. Thinks looks good.

**Comment by r-gaia-cs** [comment]
I'm merging this issue since student application period is over.

**Pull request merged**